From 8865e33025c270f4605a1079e06feb8dfd7e44ef Mon Sep 17 00:00:00 2001
From: Ondrej Holy <oholy@redhat.com>
Date: Wed, 19 Jul 2017 16:34:10 +0200
Subject: [PATCH 03/13] goa: Fix password-based authentication

Password-based authentication used for ownCloud/Nextcloud was broken
by GTask port (commit fdda284). GTask is used as user_data for async
callbacks currently, however, user_data is incorrectly cast to MountOp
in this case. Use g_task_get_task_data in order to fix this issue.

https://bugzilla.gnome.org/show_bug.cgi?id=780496
---
 monitor/goa/goavolume.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/monitor/goa/goavolume.c b/monitor/goa/goavolume.c
index 00c7e541..c077dd94 100644
--- a/monitor/goa/goavolume.c
+++ b/monitor/goa/goavolume.c
@@ -105,7 +105,7 @@ mount_operation_ask_password_cb (GMountOperation   *op,
                                  GAskPasswordFlags  flags,
                                  gpointer           user_data)
 {
-  MountOp *data = user_data;
+  MountOp *data = g_task_get_task_data (user_data);
 
   g_mount_operation_set_password (data->mount_operation, data->passwd);
   g_mount_operation_reply (data->mount_operation, G_MOUNT_OPERATION_HANDLED);
-- 
2.15.1

